onecodex/viz/_bargraph.py

```diff
@@ -134,82 +152,93 @@ def plot_bargraph(
-                if md_val is None:
-                    plot_df = df.where(df[magic_fields[haxis]].isnull()).dropna(how="all")
-                else:
-                    plot_df = df.where(df[magic_fields[haxis]] == md_val).dropna(how="all")
+            kwargs["column"] = haxis
+
+        domain = sorted(df["tax_name"].unique())
+
+        tableau10 = [
+            "#4e79a7",
+            "#f28e2b",
+            "#e15759",
+            "#76b7b2",
+            "#59a14f",
+            "#edc948",
+            "#b07aa1",
+            "#ff9da7",
+            "#9c755f",
+            "#bab0ac",
+        ]
+        other_colour = ["#d0dadb"]
+        no_level_colour = ["#eeefe1"]
 
-                # preserve booleans
-                if magic_metadata[magic_fields[haxis]].dtype == "bool":
-                    plot_df[magic_fields[haxis]] = plot_df[magic_fields[haxis]].astype(bool)
+        colour_range = (
+            tableau10 * (len(domain) // len(tableau10)) + tableau10[: len(domain) % len(tableau10)]
+        )
 
-                dfs_to_plot.append(plot_df)
-        else:
-            dfs_to_plot.append(df)
-
-        charts = []
-
-        for plot_num, plot_df in enumerate(dfs_to_plot):
-            if sort_x:
-                sort_order = sort_x(plot_df["Label"].tolist())
-
-            chart = (
-                alt.Chart(plot_df)
-                .mark_bar()
-                .encode(
-                    x=alt.X("Label", axis=alt.Axis(title=xlabel), sort=sort_order),
-                    y=alt.Y(
-                        df.ocx.field,
-                        axis=alt.Axis(title=ylabel),
-                        scale=alt.Scale(domain=[0, 1], zero=True, nice=False),
-                    ),
-                    color=alt.Color("tax_name", legend=alt.Legend(title=legend)),
-                    tooltip=tooltip_for_altair,
-                    href="url:N",
-                )
+        no_level_name = "No {}".format(rank)
+        if include_taxa_missing_rank and no_level_name in domain:
+            domain.remove(no_level_name)
+            domain = [no_level_name] + domain
+            colour_range = no_level_colour + colour_range
+
+        if include_other:
+            domain.remove("Other")
+            domain = ["Other"] + domain
+            colour_range = other_colour + colour_range
+
+        sort_order = sort_helper(sort_x, df["Label"].tolist())
+
+        df["order"] = df["tax_name"].apply(domain.index)
+
+        y_scale_kwargs = {"zero": True, "nice": False}
+        if normalize:
+            y_scale_kwargs["domain"] = [0, 1]
+
+        chart = (
+            alt.Chart(df)
+            .mark_bar()
+            .encode(
+                x=alt.X("Label", axis=alt.Axis(title=xlabel), sort=sort_order),
+                y=alt.Y(
+                    self.metric, axis=alt.Axis(title=ylabel), scale=alt.Scale(**y_scale_kwargs)
+                ),
+                color=alt.Color(
```

Comment:
sadly it uses the wrong spelling here :(

Reply:
I know :(